pkg/reconciler/route/resources/ingress.go

```diff
@@ -154,8 +154,9 @@ func MakeIngressSpec(
 	}
 
 	return netv1alpha1.IngressSpec{
-		Rules: rules,
-		TLS:   tls,
+		Rules:      rules,
+		TLS:        tls,
+		Visibility: netv1alpha1.IngressVisibilityExternalIP,
```

Comment:
I am wondering that we can delete `spec.visibility` completely from https://github.com/knative/serving/blob/master/vendor/knative.dev/networking/pkg/apis/networking/v1alpha1/ingress_types.go#L115. It seems that we are using `spec.rules.visibility` instead.

Reply:
It used to set here https://github.com/knative/serving/commit/b0c2013b48e86d9ebefb5f5a9c773d0c1cee1c0f#diff-7aca98cf2c25eb7a257bbe27cdfc7682L136

@tcnghia Do you know if we have side-effect if we removed `spec.visibility` from [IngressSpec](https://github.com/knative/serving/blob/master/vendor/knative.dev/networking/pkg/apis/networking/v1alpha1/ingress_types.go#L115)?

Reply:
Yes, we could remove that from IngressSpec. It is a fallback when `spec.rules.visibility` isn't set, but in route-controller we should be always setting it.

Reply:
Do it has side-effect for existed ingress if we removed spec.visibility.